File: ruby/yapay_gw/lib/yapay_gw/models/transaction_data.rb
class TransactionData
    attr_accessor :transaction_number
    attr_accessor :value
    attr_accessor :discount_value
    attr_accessor :boarding_fee
    attr_accessor :installments
    attr_accessor :installment_type
    attr_accessor :hint_url
    attr_accessor :result_url
    attr_accessor :unpaid_redirect_url
    attr_accessor :idiom
    attr_accessor :ip
    attr_accessor :browser
    attr_accessor :transaction_origin
    attr_accessor :free_field_one
    attr_accessor :free_field_two
    attr_accessor :free_field_three
    attr_accessor :free_field_four
    attr_accessor :free_field_five
    attr_accessor :bill_due_date
    attr_accessor :country
    attr_accessor :currency
    attr_accessor :recurrent_payment


    def as_json(options = {})
        json = {:numeroTransacao => transaction_number, 
                :valor => value,
                :valorDesconto => discount_value,
                :taxaEmbarque => boarding_fee,
                :parcelas => installments,
                :tipoParcelamento => installment_type,
                :urlCampainha => hint_url,
                :urlResultado => result_url,
                :urlRedirecionamentoNaoPago => unpaid_redirect_url,
                :idioma => idiom,
                :ip => ip,
                :browser => browser,
                :origemTransacao => transaction_origin,
                :campoLivre1 => free_field_one,
                :campoLivre2 => free_field_two,
                :campoLivre3 => free_field_three,
                :campoLivre4 => free_field_four,
                :campoLivre5 => free_field_five,
                :dataVencimentoBoleto => bill_due_date,
                :pais => country,
                :moeda => currency,
                :pagamentoRecorrente => recurrent_payment
            }
        json.delete_if { |k, v| v.nil? }
        json
    end
end

File: dotnet/YapayGatewayLib/model/TransactionData.cs
using System;
using Newtonsoft.Json;

namespace YapayGatewayLib.Model
{
    public class TransactionData
    {
        [JsonProperty(PropertyName = "numeroTransacao")]
        public long transactionNumber { get; set; }

        [JsonProperty(PropertyName = "valor")]
        public long value { get; set; }

        [JsonProperty(PropertyName = "valorDesconto")]
        public long discountValue { get; set; }

        [JsonProperty(PropertyName = "taxaEmbarque")]
        public long boardingFee { get; set; }

        [JsonProperty(PropertyName = "parcelas")]
        public int installments { get; set; }

        [JsonProperty(PropertyName = "tipoParcelamento")]
        public string installmentType { get; set; }

        [JsonProperty(PropertyName = "urlCampainha")]
        public string hintUrl { get; set; }

        [JsonProperty(PropertyName = "urlResultado")]
        public string resultUrl { get; set; }

        [JsonProperty(PropertyName = "urlRedirecionamentoNaoPago")]
        public string unpaidRedirectUrl { get; set; }

        [JsonProperty(PropertyName = "idioma")]
        public int idiom { get; set; }

        [JsonProperty(PropertyName = "ip")]
        public string ip { get; set; }

        [JsonProperty(PropertyName = "browser")]
        public string browser { get; set; }

        [JsonProperty(PropertyName = "origemTransacao")]
        public int transactionOrigin { get; set; }

        [JsonProperty(PropertyName = "campoLivre1")]
        public string freeFieldOne { get; set; }

        [JsonProperty(PropertyName = "campoLivre2")]
        public string freeFieldTwo { get; set; }

        [JsonProperty(PropertyName = "campoLivre3")]
        public string freeFieldThree { get; set; }

        [JsonProperty(PropertyName = "campoLivre4")]
        public string freeFieldFour { get; set; }

        [JsonProperty(PropertyName = "campoLivre5")]
        public string freeFieldFive { get; set; }

        [JsonProperty(PropertyName = "dataVencimentoBoleto")]
        public string billDueDate { get; set; }

        [JsonProperty(PropertyName = "pais")]
        public string country { get; set; }

        [JsonProperty(PropertyName = "moeda")]
        public string currency { get; set; }

        [JsonProperty(PropertyName = "pagamentoRecorrente")]
        public bool recurrentPayment { get; set; }
    }
}
